runtime/sema/check_path_expression.go

```diff
@@ -19,29 +19,47 @@
 package sema
 
 import (
+	"regexp"
+
 	"github.com/onflow/cadence/runtime/ast"
 	"github.com/onflow/cadence/runtime/common"
 )
 
 func (checker *Checker) VisitPathExpression(expression *ast.PathExpression) ast.Repr {
 
-	ty, err := CheckPathLiteral(expression)
+	ty, err := CheckPathLiteral(
+		expression.Domain.Identifier,
+		expression.Identifier.Identifier,
+		func() ast.Range {
+			return ast.NewRangeFromPositioned(expression.Domain)
+		},
+		func() ast.Range {
+			return ast.NewRangeFromPositioned(expression.Identifier)
+		},
+	)
+
 	checker.report(err)
 
 	return ty
 }
 
-func CheckPathLiteral(expression *ast.PathExpression) (Type, error) {
+func CheckPathLiteral(domainString, identifier string, domainRangeThunk, idRangeThunk func() ast.Range) (Type, error) {
 
 	// Check that the domain is valid
-
-	domainIdentifier := expression.Domain
-
-	domain, ok := common.AllPathDomainsByIdentifier[domainIdentifier.Identifier]
+	domain, ok := common.AllPathDomainsByIdentifier[domainString]
 	if !ok {
 		return PathType, &InvalidPathDomainError{
-			ActualDomain: domainIdentifier.Identifier,
-			Range:        ast.NewRangeFromPositioned(domainIdentifier),
+			ActualDomain: domainString,
+			Range:        domainRangeThunk(),
+		}
+	}
+
+	// Check that the identifier is valid
+	isValidIdentifier := regexp.MustCompile(`^[A-Za-z_][A-Za-z0-9_]*$`).MatchString
```

Comment:
This should be refactor into a global variable. The function should not recompile the regular expression on each invocation. It is already a global variable on master, maybe the PR is missing something?

Reply:
Ah ok there was a conflict here and I resolved it the wrong way.